### PR TITLE
[runtime] Use pending exceptions in process.c.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/process.c [hunk 1]
```diff
@@ -323,57 +333,38 @@ process_get_fileversion (MonoObject *filever, gunichar2 *filename)
...
+
+				for (i = 0; i < G_N_ELEMENTS (stringtable_entries); ++i) {
+					/* No strings, so set every field to
+					 * the empty string
+					 */
+					process_set_field_string (filever,
+											  stringtable_entries [i].name,
+											  EMPTY_STRING, 0, error);
+					if (!mono_error_ok (error))
+						return;
```
Comment (lambdageek) [inline mono/metadata/process.c]: There's a `return_if_nok(error)` macro that expands to exactly `if (!mono_error_ok (error)) return`, if you want.

File: mono/metadata/process.c [hunk 2]
```diff
@@ -391,22 +382,25 @@ process_get_assembly_fileversion (MonoObject *filever, MonoAssembly *assembly)
...
 
 	/* Build a System.Diagnostics.ProcessModule with the data.
 	 */
-	item = mono_object_new_checked (domain, proc_class, &error);
-	mono_error_raise_exception (&error); /* FIXME don't raise here */
-	filever = mono_object_new_checked (domain, mono_class_get_file_version_info_class (), &error);
-	if (!mono_error_ok (&error)) goto leave;
+	item = mono_object_new_checked (domain, proc_class, error);
+	if (!mono_error_ok (error))
+		return NULL;
```
Comment (lambdageek) [inline mono/metadata/process.c]: Also `return_val_if_nok (error, value)` that expands to `if (!mono_error_ok (error)) return value`

**Comment by lambdageek** [comment]
Other than the macros that I mentioned, if you want, lgtm.

**Comment by vargaz** [comment]
@monojenkins merge

**Pull request merged**